src/models/soil/soil_model.jl

```diff
@@ -17,33 +17,33 @@ $(TYPEDFIELDS)
     BoundaryConditions<:AbstractBoundaryConditions,
     Initializer<:AbstractInitializer,
     TimeStepper<:AbstractTimeStepper,
-} <: AbstractSoilModel{NF}
+} <: AbstractSoilModel{NF, GridType, TimeStepper}
```

Comment:
We still have this discussion in #27, do we settle this for now with explicitly handing over the precision?